runtime/js/src/main/kotlin/kotlinx/serialization/DynamicObjectParser.kt

```diff
@@ -17,8 +17,12 @@
 package kotlinx.serialization
 
 import kotlinx.serialization.CompositeDecoder.Companion.READ_DONE
-import kotlinx.serialization.context.*
+import kotlinx.serialization.context.getOrDefault
 import kotlinx.serialization.internal.EnumDescriptor
+import kotlin.math.abs
+import kotlin.math.floor
+
+internal const val MAX_SAFE_INTEGER: Double = 9007199254740991.toDouble() // 2^53 - 1
```

Comment:
Suggestion: add a link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER